tools/clang/lib/AST/ASTContextHLSL.cpp

```diff
@@ -1243,6 +1243,10 @@ CXXRecordDecl *hlsl::DeclareHitObjectType(NamespaceDecl &NSDecl) {
       static_cast<int>(hlsl::IntrinsicOp::MOP_DxHitObject_MakeNop)));
   pConstructorDecl->addAttr(HLSLCXXOverloadAttr::CreateImplicit(Context));
 
+  // Shader Execution Reordering requires SM6.9
+  VersionTuple VT69 = VersionTuple(6, 9);
+  pConstructorDecl->addAttr(ConstructAvailabilityAttribute(Context, VT69));
```

Comment:
Would you be able to add an ast-dump test to verify this new attribute in the AST? Same with the attribute you add in SemaHLSL.cpp now.
Separately, I'm not sure how this attribute is useful. Does `dx::HitObject::MakeNop();` create a constructor Decl? If not, I don't see where this attribute is actually generating an error when SM < 6.9, and I suspect the other attribute you added in SemaHLSL is actually doing the work.
For example, if you were to change this VT69 to, say, VT63, would any of the errors in your tests go away? Which ones would change?

Reply:
I've removed the AvailabilityAttribute from the constructor.. it was unused.

I've added ast tests for `HitObject::MakeNop` and `MaybeReorderThread` to check for the AvailabilityAttribute as you suggested.